interp/interp.go

```diff
@@ -335,9 +337,27 @@ func (interp *Interpreter) Repl(in, out *os.File) {
 	}
 }
 
+// trapEval is a wrapper around Eval that traps a panic if the input is a terminal
+func (interp *Interpreter) trapEval(inTerminal bool, src string) (reflect.Value, error) {
+	if inTerminal {
+		defer func() {
+			if err := recover(); err != nil {
+				fmt.Println("TrapEval: ", err)
+			}
+		}()
+	}
+	return interp.Eval(src)
+}
+
+// runningInTerminal returns true only if input is a terminal
+func runningInTerminal(in *os.File) bool {
+	stat, err := in.Stat()
+	return err == nil && stat.Mode()&os.ModeCharDevice != 0
+}
+
 // getPrompt returns a function which prints a prompt only if input is a terminal
-func getPrompt(in, out *os.File) func() {
-	if stat, err := in.Stat(); err == nil && stat.Mode()&os.ModeCharDevice != 0 {
+func getPrompt(inTerminal bool, out *os.File) func() {
```

Comment:
from the linter:
```console
$ golangci-lint run
interp/interp.go:359:33: `out` can be `io.Writer` (interfacer)
func getPrompt(inTerminal bool, out *os.File) func() {
```

You have to replace the type of the function parameter by `io.Writer`. Only in the function signature.

Reply:
After looking at a broader scope, I think it may be not necessary to fix the REPL, at least not in this order. We have to perform error processing correctly first (in the core of interpreter), so the only panics which occur are the ones we want to propagate anyway, even in REPL mode. Sorry for the wasted time.

Reply:
it seems interesting to me as a first step before treating the problem at the root.
This is not enough and we will change the behavior in the future, but it can give an impression of "robustness"

Reply:
I will avoid to respond too fast, and make another mistake 😅

Reply:
another point: this allow to write `panic` in the REPL without that the command-line panic also.